test/integration/initial-dns-seedlist-discovery/initial_dns_seedlist_discovery.prose.test.ts

```diff
@@ -0,0 +1,186 @@
+import { expect } from 'chai';
+import * as dns from 'dns';
+import * as sinon from 'sinon';
+
+import { MongoAPIError, Server, ServerDescription, Topology } from '../../mongodb';
+import { topologyWithPlaceholderClient } from '../../tools/utils';
+
+describe(
+  'Initial DNS Seedlist Discovery (Prose Tests)',
+  { requires: { topology: 'single' } },
+  () => {
+    context('When running validation on an SRV string before DNS resolution', function () {
+      beforeEach(async function () {
+        // this fn stubs DNS resolution to always pass - so we are only checking pre-DNS validation
+
+        sinon.stub(dns.promises, 'resolveSrv').callsFake(async () => {
+          return [
+            {
+              name: 'resolved.mongodb.localhost',
+              port: 27017,
+              weight: 0,
+              priority: 0
+            }
+          ];
+        });
+
+        sinon.stub(dns.promises, 'resolveTxt').callsFake(async () => {
+          throw { code: 'ENODATA' };
+        });
+
+        sinon.stub(Topology.prototype, 'selectServer').callsFake(async () => {
+          return new Server(
+            topologyWithPlaceholderClient([], {} as any),
+            new ServerDescription('a:1'),
+            {} as any
+          );
+        });
+      });
+
+      afterEach(async function () {
+        sinon.restore();
+      });
+
+      it('do not error on an SRV because it has one domain level', async function () {
+        const client = await this.configuration.newClient('mongodb+srv://localhost', {});
+        client.connect();
+        client.close();
+      });
+
+      it('do not error on an SRV because it has two domain levels', async function () {
+        const client = await this.configuration.newClient('mongodb+srv://mongodb.localhost', {});
+        client.connect();
+        client.close();
+      });
+    });
+
+    context(
+      'When given a host from DNS resolution that does NOT end with the original SRVs domain name',
```

Comment:
Can we add the number of the prose test. Makes it quicker to map from our test implementation to the test in the spec

Reply:
Yep, added them in!